experiment/build/generate_cloudbuild.py

```diff
@@ -45,33 +86,54 @@ def create_cloud_build_spec(buildable_images, docker_registry):
         if 'depends_on' in image:
             step['wait_for'] = []
             for dep in image['depends_on']:
-                step['wait_for'] += [dep]
-        cloud_build_spec['images'].append(name)
+                if 'base' in dep and not base:
+                    continue
+                step['wait_for'] += [_identity(dep)]
+            if len(step['wait_for']) == 0:
+                del step['wait_for']
+
+        cloud_build_spec['images'].append(
+            os.path.join(EXPERIMENT_TAG, image['tag']) + ':' + EXPERIMENT_VAR)
+        cloud_build_spec['images'].append(
+            os.path.join(EXPERIMENT_TAG, image['tag']))
         cloud_build_spec['steps'].append(step)
 
+    if any('coverage' in _ for _ in images_template.keys()):
+        cloud_build_spec['steps'] += coverage_steps()
+
     return cloud_build_spec
 
 
-def main():
-    """Generates Cloud Build specification."""
-    parser = argparse.ArgumentParser(description='GCB spec generator.')
-    parser.add_argument('-r',
-                        '--docker-registry',
-                        default='gcr.io/fuzzbench/',
-                        help='Docker registry to use.')
-    args = parser.parse_args()
-
-    # TODO(Tanq16): Create fuzzer/benchmark list dynamically.
-    fuzzers = ['afl', 'libfuzzer']
-    benchmarks = ['libxml', 'libpng']
-    buildable_images = docker_images.get_images_to_build(fuzzers, benchmarks)
-    cloud_build_spec = create_cloud_build_spec(buildable_images,
-                                               args.docker_registry)
-    # Build spec can be yaml or json, use whichever:
-    # https://cloud.google.com/cloud-build/docs/configuring-builds/create-basic-configuration
-    print(yaml.dump(cloud_build_spec))
-    print(json.dumps(cloud_build_spec))
-
-
-if __name__ == '__main__':
-    main()
+def _get_buildable_images():
+    return docker_images.get_images_to_build([FUZZER_VAR], [BENCHMARK_VAR])
+
+
+def generate_base_images_build_spec():
+    """Returns build spec for base images."""
+    buildable_images = _get_buildable_images()
+    images_template = {}
+    for name in buildable_images:
+        if 'base' in name:
+            images_template[name] = buildable_images[name]
+    return create_cloud_build_spec(images_template, base=True)
+
+
+def generate_benchmark_images_build_spec():
+    """Returns build spec for standard benchmarks."""
+    buildable_images = _get_buildable_images()
+    images_template = {}
+    for name in buildable_images:
+        if any(_ in name for _ in ('base', 'coverage', 'dispatcher')):
+            continue
+        images_template[name] = buildable_images[name]
+    return create_cloud_build_spec(images_template)
+
+
+def generate_coverage_images_build_spec():
+    """Returns build spec for OSS-Fuzz benchmarks."""
+    buildable_images = _get_buildable_images()
+    images_template = {}
```

Comment:
Use a dictionary comprehension for this
`images_template = {name: image for name, image in buildable_images.items() if 'coverage' in name}